### PR TITLE
MULE-12150: Change logger library and use an internal logger per test class

### PR DESCRIPTION
**Pull request by manuporto**
Fixes `java.lang.IllegalStateException: Lifecycle Manager 'MuleContext' phase 'start' does not support phase 'dispose'`

### CODE REVIEW

**Comment by manuporto** [comment]
`java.security.SignatureException: Invalid encoding for signature` it's still not solved

**Comment by manuporto** [comment]
This changes makes MultipleConnectorsMule1765TestCase fail. Do not merge until solved

**Comment by manuporto** [comment]
Bug solved

File: core/src/test/java/org/mule/tck/junit4/AbstractMuleContextTestCase.java [hunk 1]
```diff
@@ -320,7 +320,14 @@ public static void disposeContext()
         {
             if (muleContext != null && !(muleContext.isDisposed() || muleContext.isDisposing()))
             {
-                muleContext.dispose();
+                try
+                {
+                    muleContext.dispose();
+                } catch (IllegalStateException e)
```
Comment (elrodro83) [inline core/src/test/java/org/mule/tck/junit4/AbstractMuleContextTestCase.java]: format

File: core/src/test/java/org/mule/tck/junit4/AbstractMuleContextTestCase.java [hunk 2]
```diff
@@ -320,7 +320,14 @@ public static void disposeContext()
...
+                {
+                    // Ignore
```
Comment (elrodro83) [inline core/src/test/java/org/mule/tck/junit4/AbstractMuleContextTestCase.java]: log a warning at least

File: core/src/test/java/org/mule/tck/junit4/AbstractMuleContextTestCase.java [hunk 3]
```diff
@@ -320,7 +323,16 @@ public static void disposeContext()
...
             {
-                muleContext.dispose();
+                try
+                {
+                    muleContext.dispose();
+                }
+                catch (IllegalStateException e)
+                {
+                    // Ignore
+                    logger.warn(e.getMessage());
```
Comment (elrodro83) [inline core/src/test/java/org/mule/tck/junit4/AbstractMuleContextTestCase.java]: log the exception too, not just the messgae

File: core/src/test/java/org/mule/tck/junit4/AbstractMuleTestCase.java [hunk 4]
```diff
@@ -6,41 +6,28 @@
...
-import org.junit.BeforeClass;
-import org.junit.Rule;
+import org.junit.*;
 import org.junit.rules.DisableOnDebug;
 import org.junit.rules.TestName;
 import org.junit.rules.TestRule;
 import org.junit.rules.Timeout;
+import org.mule.RequestContext;
+import org.mule.tck.junit4.rule.WarningTimeout;
+import org.mule.util.*;
```
Comment (elrodro83) [inline core/src/test/java/org/mule/tck/junit4/AbstractMuleTestCase.java]: configure your IDE to not use * imports

File: transports/tcp/src/test/java/org/mule/transport/tcp/other/SocketTimingExperimentTestCase.java [hunk 5]
```diff
@@ -38,7 +40,7 @@ public void testSocketTiming() throws IOException, InterruptedException
         }
         catch (Exception e)
         {
-            logger.info(e);
+            logger.info(String.valueOf(e));
```
Comment (elrodro83) [inline transports/tcp/src/test/java/org/mule/transport/tcp/other/SocketTimingExperimentTestCase.java]: log the exception as exception, not as string

**Comment by aiannucci** [comment]
@elrodro83 can we merge this?

**Comment by elrodro83** [comment]
jenkins validate

**Pull request merged**